src/tools/rbd/action/Lock.cc

```diff
@@ -26,13 +26,15 @@ void add_id_option(po::options_description *positional) {
 
 int get_id(const po::variables_map &vm, size_t *arg_index,
            std::string *id) {
-  *id = utils::get_positional_argument(vm, *arg_index);
-  if (id->empty()) {
+  if (vm.count(at::POSITIONAL_ARGUMENTS) == 0 ||
+      *arg_index >= boost::any_cast<std::vector<std::string>>(
+        vm[at::POSITIONAL_ARGUMENTS].value()).size()) {
```

Comment:
Repeating these checks here breaks the abstraction between argument processing and action logic.  How about adding `get_positional_argument()` overload that would return whether the argument was specified?  `get_id()` would use the new overload and check the return value, the current behavior can be preserved by ignoring the return value.  I'm thinking of something like
```
bool get_positional_argument(const po::variables_map& vm, size_t index,
                             std::string* arg) {
  if (vm.count(at::POSITIONAL_ARGUMENTS) == 0) {
    return false;
  }

  const std::vector<std::string> &args =
    boost::any_cast<std::vector<std::string> >(
      vm[at::POSITIONAL_ARGUMENTS].value());
  if (index >= args.size()) {
    return false;
  }
  *arg = args[index];
  return true;
}

std::string get_positional_argument(const po::variables_map &vm, size_t index) {
  std::string arg;
  get_positional_argument(vm, index, &arg);
  return arg;
}
```
in `src/tools/rbd/Utils.cc`.